plugins/modules/digital_ocean_floating_ip.py

```diff
@@ -294,17 +294,19 @@ def create_floating_ips(module, rest):
 
     # Exit unchanged if any of them are assigned to this Droplet already
     if status_code == 200:
-        for floating_ip in json_data.get('floating_ips', []):
-            droplet = floating_ip.get('droplet', None)
-            if droplet is not None:
-                droplet_id = droplet.get('id', None)
-                if droplet_id is not None:
-                    if droplet_id == module.params['droplet_id']:
-                        ip = floating_ip.get('ip', None)
-                        if ip is not None:
-                            module.exit_json(changed=False, data={'floating_ip': ip})
-                        else:
-                            module.fail_json(changed=False, msg="Unexpected error querying floating ip")
+        floating_ips = json_data.get('floating_ips', [])
+        if floating_ips != []:
```

Comment:
I think "this is fine", but how to check for an empty list is a subject of [immense debate](https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty) in the Python community. =)

You might consider doing:

```
if floating_ips:
```

This relies on boolean coercion rules, such as:
```
>>> bool([])
False
>>> bool(['127.0.0.1'])
True
```

Or you could do:
```
if len(floating_ips) != 0:
```

Reply:
I went with the last suggestion, thank you!